### DISCUSSION TITLE
Help to configure yuhaiin configuration

### DISCUSSION THREAD

**Discussion by Vicio59** [yuhaiin/yuhaiin#197] [Q&A]
Hi, I'm looking for a proxy client that supports Full Cone NAT and can run on linux, in my case I wanted to use a docker container that runs alpine linux. I would need help (if possible step-by-step for the configuration), I found this project very interesting, in my case I want to route all inbound traffic to a tun interface and outbound traffic to a socks5 server which requires username and password as an authentication method. Thanks for help

**Comment by Asutorufa** [comment]
Hi, Vicio59, if you want use tun, I recommend run binary with root instead of run with docker.

Here is my method in my linux pc.

first, run yuhaiin with

```shell
sudo yuhaiin -host="0.0.0.0:50051" -path=${HOME}/.config/yuhaiin
```

1. then open browser and access http://127.0.0.1:50051
2. add new socks 5 config at `group -> New`, here is an example
![image](https://github.com/yuhaiin/yuhaiin/assets/16442314/d57d05b1-f074-48f4-ab38-a7268b74578a)
test button can test the node
![image](https://github.com/yuhaiin/yuhaiin/assets/16442314/d68c0709-80fe-4095-a902-d0173329ce04)
3. then change to use this new node
![image](https://github.com/yuhaiin/yuhaiin/assets/16442314/5b596094-b6d5-4bcb-be16-3ba0261f750b)
4. config fakedns at config
![image](https://github.com/yuhaiin/yuhaiin/assets/16442314/51778b8f-c85b-48b0-8fdb-ea14490dd6f3)
5. enabled dnshijack and fakedns at inbound
![image](https://github.com/yuhaiin/yuhaiin/assets/16442314/8c7d6a42-af1a-48f3-8125-e06f49b0dfb8)
6. finally add tun inbound and enabled, add fake dns range to route, some application(eg: telegram) use ip instead of domain, you can also add these ip range to route.
![image](https://github.com/yuhaiin/yuhaiin/assets/16442314/dce1f2f1-71d1-4b12-8b5b-10bf0279eb98)

**Reply by Vicio59** [reply]
Hi, thanks for your precious help, I started configuring yuhaiin, but I encountered a strange problem. I added this configuration on group:
![Screenshot (952)](https://github.com/yuhaiin/yuhaiin/assets/156183766/ba74cfd6-84fd-472a-8565-09b44761e850)

Where PUBLICSERVERIP is the IP address of another vps where I run the socks5 server.
Then I click on the test button and only TCP works with a ping of 20-50 ms while for UDP I receive a timeout even though my socks5 server supports UDP. The problem is that when I subsequently click on: USE and select TCP, the test no longer works for either TCP or UDP. also I tried this command: curl -v 1.1.1.1 --interface tun0 to test TCP connectivity but it doesn't work and I get this:
* Trying 1.1.1.1:80...
* socket successfully bound to interface 'tun0'

I remember that my intent is only to route all traffic on the tun0 interface to change my public IP with that of the socks5 server which is located in another vps. The server works perfectly because I tested it with both tun2socks and tun2proxy, I would like to use yuhaiin because it supports Full cone nat. Thanks for your patience.

**Reply by Asutorufa** [reply]
Hi Vicio59.
1. The socks5 hostname don't need port, you also need set the simple host to the public server ip. tbh the socks5 hostname only used when the socks5 server return udp server address is 0.0.0.0.
2. If you want route all traffic, I assume you set the tun inbound route is 0.0.0.0/0, in this case you maybe need set the Network Interface to the internet interface, otherwise the network will infinite loop by the tun0.

**Comment by Vicio59** [comment]
Hi, thanks again for your support, I configured the group like this but I keep getting timeouts during the test, what am I doing wrong?
![Screenshot (955)](https://github.com/yuhaiin/yuhaiin/assets/156183766/dadf8b5a-7791-4906-815b-e53b3729b247)

**Reply by Asutorufa** [reply]
Hi, Vicio59, i have found some typo at socks5 client auth password, can you try this version https://github.com/yuhaiin/yuhaiin/actions/runs/9361650734.
Socks5 hostname should same as the host of simple.

**Comment by Vicio59** [comment]
Hi, thank you again for your support, I confirm that the ping problem has been resolved, now by clicking the test button I receive the ping for both TCP and UDP.
![Screenshot (958)](https://github.com/yuhaiin/yuhaiin/assets/156183766/c7cdca47-99f9-4699-a760-d14c5ba01975)

The problem is that if I run curl or traceroute, it still appears that the traffic is not going through the proxy in fact these tests fail. These are the test commands I did:
traceroute -i tun0 1.1.1.1
curl -v --interface tun0 1.1.1.1
These are screenshots of my setup:
![Screenshot (959)](https://github.com/yuhaiin/yuhaiin/assets/156183766/0702720a-2d43-4450-a4ec-e76c1b69c5d8)
![Screenshot (960)](https://github.com/yuhaiin/yuhaiin/assets/156183766/a7ed50b7-9c6d-4bf6-a4f7-e19cf27e66dd)
![Screenshot (961)](https://github.com/yuhaiin/yuhaiin/assets/156183766/e5b5277f-b529-455c-9f4d-320d74349250)
![Screenshot (963)](https://github.com/yuhaiin/yuhaiin/assets/156183766/f5b02d1a-ea17-4a7a-b8aa-acb94a5d70bc)
![Screenshot (964)](https://github.com/yuhaiin/yuhaiin/assets/156183766/bf8e65ce-55c8-4d87-95e6-ea4b9d55147e)

**Reply by Asutorufa** [reply]
current you use fakedns, so you need set system dns to 172.19.0.2 and fdfe:dcba:9876::2.
then try `curl -v --interface tun0 one.one.one.one` or `curl -v --interface tun0 ip.sb`, if you want route 1.1.1.1 to tun, you need add 1.1.1.1/32 to tun routes.

you can use dig to check is dns set successful.
```shell
$ dig ip.sb

; <<>> DiG 9.18.27 <<>> ip.sb
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 24216
;; flags: qr; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;ip.sb.                         IN      A

;; ANSWER SECTION:
ip.sb.                  600     IN      A       10.0.0.2  <--------------- the dns return ip is fakeip

;; Query time: 0 msec
;; SERVER: 172.19.0.2#53(172.19.0.2) (UDP)
;; WHEN: Tue Jun 04 22:47:03 CST 2024
;; MSG SIZE  rcvd: 39
```

**Reply by Asutorufa** [reply]
system dns sample

```shell
$ cat /etc/resolv.conf
# Generated by NetworkManager
nameserver 172.19.0.2
nameserver fdfe:dcba:9876::2
```

**Reply by Asutorufa** [reply]
if you use kde, dns should set at system settings.

![image](https://github.com/yuhaiin/yuhaiin/assets/16442314/83400062-8bce-4539-9188-8662ff35231e)

**Comment by Vicio59** [comment]
Hi, now it works perfectly, I think I understand how it works, if possible I would like to ask you two questions:
1. I tried to route all the traffic by adding routes 0.0.0.0/0 and even though I added "Network Interface: eth0" the loop still seems to exist. How can I route all traffic through tun0 without creating the loop.
2. What is the difference between direct DNS (for example Google's) and FakeDNS? For example, for my purpose (which is just to change the public IP of my servers), which DNS do you recommend me to use?

**Reply by Asutorufa** [reply]
1. Adjusting routing is a complex task depending on network conditions. I think you should learn about linux route(eg: ip route cli) and iptables.
2. Fake dns is just a method route the domain to tun. it works like:

    - when application resolve domain(eg: www.google.com), yuhaiin(or some other proxy tools) return fakeip (eg: 10.0.0.2).
    - system route 10.0.0.2 to tun.
    - yuhaiin convert fakeip(eg: 10.0.0.2) to domain(eg: www.google.com).
    - then send the domain(eg: www.google.com) to remote proxy server.

3. So if you can resolve the first problem(route all the traffic), you can just use some public dns, otherwise I still recommend using fakedns and route ip manual.

**Reply by Vicio59** [reply]
Hi, I have been using your yuhaiin proxy client for a few months and it works quite well, really congratulations for your professionalism and for dedicating your time last time to help me use this proxy client. As I told you it works very well but with my proprietary Linux software the TCP requests seem to be closed very quickly. This is the yuhaiin log that always repeats like this:

```go
time=2024-07-15T04:55:58.853Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:55:58.860Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:55:58.861Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:55:58.873Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:55:58.881Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:55:58.881Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:55:58.886Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:55:58.890Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:55:58.890Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:55:58.893Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:56:15.283Z level=DEBUG source=statistic.go:100 msg="new conn" id=266238 addr=(DESTINATION-IP):443 src=172.19.0.1:59861 network=tcp outbound=(PUB-IP-SERVER-PROXY):1080
time=2024-07-15T04:56:15.830Z level=DEBUG source=statistic.go:91 msg="close conn" id=266238
time=2024-07-15T04:56:39.248Z level=DEBUG source=statistic.go:100 msg="new conn" id=266239 addr=(DESTINATION-IP):443 src=172.19.0.1:44269 network=tcp outbound=(PUB-IP-SERVER-PROXY):1080
time=2024-07-15T04:56:39.947Z level=DEBUG source=statistic.go:91 msg="close conn" id=266239
time=2024-07-15T04:56:46.980Z level=DEBUG source=statistic.go:100 msg="new conn" id=266240 addr=(DESTINATION-IP):443 src=172.19.0.1:45917 network=tcp outbound=(PUB-IP-SERVER-PROXY):1080
time=2024-07-15T04:56:47.855Z level=DEBUG source=statistic.go:91 msg="close conn" id=266240
time=2024-07-15T04:57:59.109Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:57:59.114Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:57:59.115Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:57:59.124Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:57:59.130Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:57:59.134Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:57:59.136Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:57:59.139Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:57:59.143Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:57:59.144Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T04:57:59.634Z level=DEBUG source=statistic.go:100 msg="new conn" id=266238 addr=(DESTINATION-IP):443 src=172.19.0.1:59861 network=tcp outbound=(PUB-IP-SERVER-PROXY):1080
time=2024-07-15T04:57:59.634Z level=DEBUG source=statistic.go:91 msg="close conn" id=266241
time=2024-07-15T04:58:19.381Z level=DEBUG source=statistic.go:100 msg="new conn" id=266238 addr=(DESTINATION-IP):443 src=172.19.0.1:59861 network=tcp outbound=(PUB-IP-SERVER-PROXY):1080
time=2024-07-15T04:58:20.006Z level=DEBUG source=statistic.go:91 msg="close conn" id=266242
time=2024-07-15T04:58:50.524Z level=DEBUG source=statistic.go:100 msg="new conn" id=266238 addr=(DESTINATION-IP):443 src=172.19.0.1:59861 network=tcp outbound=(PUB-IP-SERVER-PROXY):1080
time=2024-07-15T04:58:51.057Z level=DEBUG source=statistic.go:91 msg="close conn" id=266243
time=2024-07-15T04:59:21.556Z level=DEBUG source=statistic.go:100 msg="new conn" id=266238 addr=(DESTINATION-IP):443 src=172.19.0.1:59861 network=tcp outbound=(PUB-IP-SERVER-PROXY):1080
time=2024-07-15T04:59:22.083Z level=DEBUG source=statistic.go:91 msg="close conn" id=266244
time=2024-07-15T05:00:00.126Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T05:00:00.126Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T05:00:00.162Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T05:00:00.162Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T05:00:00.163Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T05:00:00.167Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T05:00:00.171Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T05:00:00.173Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
time=2024-07-15T05:00:00.174Z level=ERROR source=handler.go:48 msg=stream error="dial (DESTINATION-IP):4277 failed: connect tcp to (DESTINATION-IP):4277 failed: dial (DESTINATION-IP):4277 failed: dial failed: simple dial failed: errTimeout"
``` 

Is it possible that it's a TCPkeepalive problem? The software works great without using proxy. Thanks in advance for your support

**Reply by Asutorufa** [reply]
I think it is network problem, maybe sometime can't connect to proxy server?
can you try latest version https://github.com/yuhaiin/yuhaiin/actions/runs/9924974978 ?

**Reply by Vicio59** [reply]
Hi, I tried the new version but I'm encountering the same problem, these days I've done various tests and I've discovered a very strange behavior, if I run yuhaiin it works perfectly, and I can also run a speedtest-cli reaching a very good speed, last test : ping: 58 ms, download: 1718.64 Mbps and upload: 1166.38Mbps, and I can also execute 5 or 6 curl commands in the terminal (curl ifconfig.me), at a certain point I can't execute anything anymore because it seems that yuhaiin freezes , to unlock it I found that I need to run this command:
```go
ip rule delete from all lookup 63
ip rule add from all lookup 63 prio 32765
``` 
magically yuhaiin starts working again but after a short while (the time to run the speedtest and curl tests) the same problem returns again.

This is my ip rule configuration:
```go
0:      from all lookup local
32756:  from <PUB-IP-SERVER-PROXY> lookup main
32757:  from all to <PUB-IP-SERVER-PROXY> lookup main
32760:  from all iif lo ipproto tcp sport 50051 lookup main
32765:  from all lookup 63
32766:  from all lookup main
32767:  from all lookup default
``` 
![Screenshot (992)](https://github.com/user-attachments/assets/ace07f92-9d94-43b1-8223-a91ee0fd55d2)

On tun0 I use 0.0.0.0/0 to route all traffic through the proxy, this explains why I use the following ip rule:
```go
ip rule from <PUB-IP-SERVER-PROXY> lookup main
ip rule from all to <PUB-IP-SERVER-PROXY> lookup main
```

**Reply by Asutorufa** [reply]
Thank you for your feedback, i will try it on my own archlinux pc when I have time.
btw can you give me the logs, the location at `/root/.config/yuhaiin/log/` or `~/.config/yuhaiin/log/`.

**Reply by Asutorufa** [reply]
Hi, i tried it by myself, looks like no problem for me.
Can you try change the tun stack from `system` to `channel`.

Secondly, you can try save the following cidr(excluding private address ranges) to a files
```
0.0.0.0/5
8.0.0.0/7
11.0.0.0/8
12.0.0.0/6
16.0.0.0/4
32.0.0.0/3
64.0.0.0/3
96.0.0.0/6
100.0.0.0/10
100.128.0.0/9
101.0.0.0/8
102.0.0.0/7
104.0.0.0/5
112.0.0.0/4
128.0.0.0/3
160.0.0.0/5
168.0.0.0/6
172.0.0.0/12
172.32.0.0/11
172.64.0.0/10
172.128.0.0/9
173.0.0.0/8
174.0.0.0/7
176.0.0.0/4
192.0.0.0/9
192.128.0.0/11
192.160.0.0/13
192.169.0.0/16
192.170.0.0/15
192.172.0.0/14
192.176.0.0/12
192.192.0.0/10
193.0.0.0/8
194.0.0.0/7
196.0.0.0/6
200.0.0.0/5
208.0.0.0/4
::/1
8000::/2
c000::/3
e000::/4
f000::/5
f800::/6
fe00::/9
fec0::/10
```
and just route these ranges
![image](https://github.com/user-attachments/assets/af7ae46b-c6f4-4a42-abb6-b9764fbdad6c)

Finally I try to change some dialer configs, can you try this version https://github.com/yuhaiin/yuhaiin/actions/runs/10025428862 .

**Comment by Vicio59** [comment]
Hi, the last problem from last time I finally solved it by reinstalling the Ubuntu operating system, I don't know why but it worked, I always wanted to thank you for the help you give. This time the problem is a bit different, with some socks5 proxy servers that I have udp does not work with the latest version of yuhaiin, and I receive N/A as in the image below on UDP, while TCP works perfectly.
![Screenshot (1053)](https://github.com/user-attachments/assets/8bdf41fd-7db6-4348-bbf6-227b9a8fc418)


In old versions of yuhaiin i get a timeout error. 
![Screenshot (1048)](https://github.com/user-attachments/assets/de15aee5-8fb1-4c03-be5a-b2ee5fd9b799)


The same socks5 proxy servers however work with both TCP and UDP if I use other proxy tools, I tried with windows instead of linux the tools: sockscap64 and Netch, and UDP works, what can I do? Thanks in advance for everything so far.
![Screenshot (1054)](https://github.com/user-attachments/assets/4cdc4bba-8012-45ae-89db-02b1e09a60c9)

**Reply by Vicio59** [reply]
I can also share with you the socks5 proxy if you want to test it, no problem

**Reply by Asutorufa** [reply]
I think it's the different testing methods, yuhaiin will send data to a real remote server which current is public dns over quic server, maybe these socks5 server can't access the default test server. And the latest code version at https://github.com/yuhaiin/yuhaiin/actions/runs/10511359482. I think you can try:
1. Change test server at WEBUI tab -> latency -> DOQ, change it to other dns servers like p0.freedns.controld.com:853.
2. You can try access some http3 website like youtube, the browser f12 tools-> network can show the protocol is http3 or others, the blow screenshot browser is firefox.
![image](https://github.com/user-attachments/assets/d5e71935-338e-4eea-91ea-8fe54a8a3230)

If above method can't resolve you problem, maybe i really need the yuhaiin logs and you socks5 server to test.

Sorry for my poor english, maybe some of the English grammar is a bit weird.

**Reply by Vicio59** [reply]
Hi thanks for your precious help, I tried these methods but it didn't work, even if I didn't understand number 2 very well. I was wondering: isn't there a way to tell yuhaiin to still use the socks5 server with the UDP protocol even if the test fails? As far as I understand if the test fails yuhaiin will not use UDP, right? Anyway as I said I am also willing to send you the credentials of this socks5 server, where could I send it to you privately?

**Reply by Asutorufa** [reply]
yuhaiin will always use the node that current used, even it is test failed.  
Can you send it by email? you can find it at my github page. I will try it in recently.

**Reply by Vicio59** [reply]
Thanks but I can't find your email on your profile, it's probably my fault 😂

**Reply by Asutorufa** [reply]
9um0f♡2429um0f9.eu.org(♡→@)

**Reply by Asutorufa** [reply]
Hi, I already receive the email, I will try it later.

**Reply by Asutorufa** [reply]
@Vicio59 Hi, I tried it, but it seems just network problem, I can't get any udp response from socks5 server, can you see any error/logs on your socks5 server?

**Reply by Vicio59** [reply]
Hi, unfortunately I don't have access to this proxy server, I only tested it with sockscap64 client and with netch on Windows
